spec/lib/krikri/harvesters/oai_harvester_spec.rb

```diff
@@ -453,6 +453,55 @@
       end
     end
 
+    describe 'identifiers' do
+      it 'uses oai header identifier by default' do
+        id = 'oai:oaipmh.huygens.knaw.nl:arthurianfiction:MAN0000000010'
+
+        expect(subject).to receive(:mint_id).with(id).and_return('id')
+        subject.records.first
+      end
+
+      context 'with :id_path' do
+        let(:args) do
+          { uri: 'http://example.org/endpoint', 
+            oai: { id_path: '//dc:identifier' } }
+        end
+        
+        it 'returns the identifier from a given xpath' do
+          id = 'https://service.arthurianfiction.org/manuscript/MAN0000000010'
+
+          expect(subject).to receive(:mint_id).with(id).and_return('id')
+          subject.records.first
+        end
+
+        context 'and bad path' do
+          let(:args) do
+            { uri: 'http://example.org/endpoint', 
+              oai: { id_path: '//dc:not_a_path' } }
+          end
+          
+          it 'falls back on the header identifier' do
```

Comment:
Is this acceptable? I don't believe the legacy ingestion system has a fallback behavior, but since `dc:identifier` is not guaranteed to be present across records/collections, it seems like we would not want to fail on any records that are missing the field.